codes/c/chapter_hashing/hash_map_chaining.c

```diff
@@ -0,0 +1,235 @@
+/**
+ * File: hash_map_chaining.c
+ * Created Time: 2023-08-29
+ * Author: SenMing (1206575349@qq.com)
+ */
+
+#include "../utils/common.h"
+
+/* 键值对 int->string */
+typedef struct node {
+    int key;
+    char* val;
+    struct node* next;              // 下一个元素
+} Node;
+
+typedef struct pair {
+    Node* node;                     // 键值对节点
+} Pair;
+
+/* 基于数组简易实现的哈希表 */
+typedef struct ArrayHashMap  {
```

Comment:
Naming needs to be consistent with other languages. such as [`HashMapChaining`](https://github.com/krahets/hello-algo/blob/8a0a8b80cf2f46f3840c81d5ab440feb71115d7b/codes/cpp/chapter_hashing/hash_map_chaining.cpp#L10C7-L10C23)